rebasehelper/utils.py

```diff
@@ -285,61 +287,56 @@ def format_size(s):
         sys.stdout.flush()
 
     @staticmethod
-    def download_file(url, destination_path, timeout=10, blocksize=8192):
+    def download_file(url, destination_path, blocksize=8192):
         """
         Method for downloading file from HTTP, HTTPS and FTP URL.
 
         :param url: URL from which to download the file
         :param destination_path: path where to store downloaded file
-        :param timeout: timeout in seconds for blocking actions like connecting, etc.
         :param blocksize: size in Bytes of blocks used for downloading the file and reporting progress
         :return: None
         """
+        requests_ftp.monkeypatch_session()
+        session = requests.Session()
+        r = session.get(url)
+        if r.status_code == 404:
+            raise DownloadError
+
+        lowercase_headers = {k.lower(): v for k, v in r.headers.items()}
```

Comment:
`r.headers` is of type `requests.structures.CaseInsensitiveDict`, so this should be unnecessary.